main.go

```diff
@@ -194,6 +197,36 @@ func main() {
 			setupLog.Error(dashErr, "failed to create the OpenShift dashboards")
 		}
 	}
+	if featuregate.EnableOperatorNetworkPolicy.IsEnabled() {
+		operatorNamespace := os.Getenv("NAMESPACE")
+		if operatorNamespace == "" {
+			setupLog.Error(errors.New("NAMESPACE environment variable is not set"), "failed to create the Operator network policies")
+			os.Exit(1)
+		}
+		var policyOpts []operatornetworkpolicy.Option
+		policyOpts = append(policyOpts, operatornetworkpolicy.WithOperatorNamespace(operatorNamespace))
+
+		if cfg.EnableWebhooks {
+			policyOpts = append(policyOpts, operatornetworkpolicy.WithWebhookPort(int32(cfg.WebhookPort)))
+		}
+		if cfg.MetricsAddr != "" {
+			_, portStr, errParse := net.SplitHostPort(cfg.MetricsAddr)
+			if errParse != nil {
+				setupLog.Error(errParse, "failed to parse port from metrics address", "metricsAddr", cfg.MetricsAddr)
+				os.Exit(1)
+			}
+			metricsPort, errParse := strconv.Atoi(portStr)
+			if errParse != nil {
+				setupLog.Error(errParse, "failed to parse port from metrics address", "metricsAddr", cfg.MetricsAddr)
+				os.Exit(1)
+			}
+			policyOpts = append(policyOpts, operatornetworkpolicy.WithMetricsPort(int32(metricsPort)))
+		}
+		operatorNetworkPoliciesErr := mgr.Add(operatornetworkpolicy.NewOperatorNetworkPolicy(clientset, mgr.GetScheme(), policyOpts...))
+		if operatorNetworkPoliciesErr != nil {
+			setupLog.Error(operatorNetworkPoliciesErr, "failed to create the Operator network policies")
+		}
+	}
```

Comment:
Since this is all port parsing for the network policies, should we move it into `internal/operatornetworkpolicy`?

Ether into the construction or as option (`func WithSettingsFromConfig(cfg config.Config) (Option, error)`)?.

Reply:
> func WithSettingsFromConfig(cfg config.Config) (Option, error))

I would prefer to keep the `networkpolicy` package without a dependency on the `config` package. I also prefer to have the options pattern operate on options that are explicitly needed by the creator function.